### PR TITLE
near/testing: improve near tests for decimal point shifts

### PR DESCRIPTION
**Pull request by jumpsiegel**
near/testing: improve near tests for decimal point shifts
                       force algorand and indexer to update to what we are about to run in production
                       add checks for fees being passed for P3 messages

### CODE REVIEW

File: algorand/sandbox-algorand/images/indexer/Dockerfile [hunk 1]
```diff
@@ -29,6 +29,7 @@ COPY images/indexer/start.sh /tmp/start.sh
 COPY images/indexer/install.sh /tmp/install.sh
 
 # Install indexer binaries.
+
```
Comment (jumpsiegel) [inline algorand/sandbox-algorand/images/indexer/Dockerfile:32]: Causes tilt to detect a change

File: algorand/sandbox-algorand/images/indexer/install.sh [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #   URL    - Git repository URL.
 #   BRANCH - Git branch to clone.
 #   SHA    - (optional) Specific commit hash to checkout.
+
```
Comment (jumpsiegel) [inline algorand/sandbox-algorand/images/indexer/install.sh:9]: Helps tilt detect the change...

File: near/contracts/ft/src/lib.rs [hunk 3]
```diff
@@ -157,6 +157,10 @@ impl FTContract {
                 env::panic_str(&format!("paylod1 formatting errro  len = {}", p.len()));
             }
         } else {
+            if fee != 0 {
+                env::panic_str("Payload3 does not support fees");
```
Comment (bruce-riley) [inline near/contracts/ft/src/lib.rs:161]: Is this only hit for Payload3, as the error indicates?

File: near/contracts/ft/src/lib.rs [hunk 4]
```diff
@@ -157,6 +157,10 @@ impl FTContract {
...
+            }
+
             p = [p, hex::decode(&payload).unwrap()].concat();
             if p.len() != (133 + (payload.len() / 2)) {
                 env::panic_str(&format!("paylod3 formatting errro  len = {}", p.len()));
```
Comment (bruce-riley) [inline near/contracts/ft/src/lib.rs:166]: Typo "paylod3".

File: near/contracts/token-bridge/src/lib.rs [hunk 5]
```diff
@@ -975,6 +981,10 @@ impl TokenBridge {
             env::panic_str("EmptyTransfer");
         }
 
+        if !payload.is_empty() && nfee != 0 {
+            env::panic_str("Payload3 does not support fees");
```
Comment (bruce-riley) [inline near/contracts/token-bridge/src/lib.rs:985]: Same comment as above.

File: near/contracts/token-bridge/src/lib.rs [hunk 6]
```diff
@@ -1498,6 +1508,10 @@ impl TokenBridge {
             env::panic_str("EmptyTransfer");
         }
 
+        if !tp.payload.is_empty() && nfee != 0 {
+            env::panic_str("Payload3 does not support fees");
```
Comment (bruce-riley) [inline near/contracts/token-bridge/src/lib.rs:1512]: Ditto.

**Review by bruce-riley — DISMISSED** [review]

File: near/test/sdk.ts [hunk 7]
```diff
@@ -520,25 +523,21 @@ async function testNearSDK() {
...
       account_id: token_bridge,
-    }
-  ));
+    })
+  );
 
   let randoTransfer;
   {
-    console.log("transfer rando token from near to algorand");
+    console.log("YYY transfer rando token from near to algorand");
```
Comment (panoel) [inline near/test/sdk.ts:540]: Is the "YYY" for easy greping?

**Review by panoel — DISMISSED** [review]

**Comment by panoel** [comment]
Are you using vscode + prettier or just like emacs?

File: near/contracts/ft/src/lib.rs (hunk 3 above)
Comment (jumpsiegel) [reply near/contracts/ft/src/lib.rs:161]: Yup, this is the P3 case..

File: near/contracts/ft/src/lib.rs (hunk 4 above)
Comment (jumpsiegel) [reply near/contracts/ft/src/lib.rs]: Fixed... amusing

File: near/contracts/token-bridge/src/lib.rs (hunk 5 above)
Comment (jumpsiegel) [reply near/contracts/token-bridge/src/lib.rs:985]: Yeah, this check is just for P3

**Review by panoel — APPROVED** [review]

**Review by panoel — APPROVED** [review]

**Review by kcsongor — APPROVED** [review]

**Pull request merged**